### PR TITLE
fix: error around latest partition in BigQuery

### PR DESCRIPTION
**Pull request by mistercrunch**
Hit this bug while using BigQ. The SQLAlchemy inspector's `get_indexes` returns a list of columns that is wrong and get the SQL Lab frontend to hard crash. 

Here's the exact payload I was getting off of `get_indexes` for a particular table:
`[{'name': 'partition', 'column_names': [None], 'unique': False}]`

Here I'm adding a BigQuery specific mutation for the indexes that makes sure this doesn't happen. We're assuming that `pybigquery` will fix the root cause in a future version.

Screen Shot 2020-10-18 at 10 27 37 PM

### CODE REVIEW

File: superset-frontend/src/SqlLab/components/TableElement.jsx [hunk 1]
```diff
@@ -110,7 +110,7 @@ class TableElement extends React.PureComponent {
           />
         );
       }
-      let latest = Object.entries(table.partitions.latest).map(
+      let latest = Object.entries(table.partitions?.latest || []).map(
```
Comment (mistercrunch) [inline superset-frontend/src/SqlLab/components/TableElement.jsx:113]: This isn't necessary anymore given the issue being addressed in the backend, but is a tiny bit of defensive programming.

File: tests/db_engine_specs/bigquery_tests.py [hunk 2]
```diff
@@ -122,6 +122,18 @@ def test_extra_table_metadata(self):
...
 
+    def test_normalize_indexes(self):
+        """
+        DB Eng Specs (bigquery): Test extra table metadata
+        """
+        indexes = [{"name": "partition", "column_names": [None], "unique": False}]
+        normalized_idx = BigQueryEngineSpec.normalize_indexes(indexes)
+        self.assertEqual(normalized_idx, [])
+
+        indexes = [{"name": "partition", "column_names": ["dttm"], "unique": False}]
```
Comment (dpgaspar) [inline tests/db_engine_specs/bigquery_tests.py:133]: we could add another small test with `indexes = [{"name": "partition", "column_names": ["dttm", None], "unique": False}]`

**Review by dpgaspar — APPROVED** [review]
LGTM

File: superset/db_engine_specs/bigquery.py [hunk 3]
```diff
@@ -129,6 +129,25 @@ def _truncate_label(cls, label: str) -> str:
...
+
+        :param indexes: Raw indexes as returned by SQLAlchemy
+        :return: cleaner, more aligned index definition
+        """
+        normalized_idxs = []
+        # Fixing a bug/behavior observed in pybigquery==0.4.15 where
+        # the index's `column_names` == [None]
+        # Here we're returning only non-None indexes
+        for ix in indexes:
+            column_names = ix.get("column_names") or []
```
[the remark this answers is not in the document]
Comment (mistercrunch) [reply superset/db_engine_specs/bigquery.py:145]: the `or` is a bit safer in case the key exists with a `None` value, will stick to it

**Comment by mistercrunch** [comment]
addressed the comments, mergin'!

**Pull request merged**